test/unit/context_unit_test.rb

```diff
@@ -466,4 +466,46 @@ def test_context_initialization_with_a_proc_in_environment
     assert contx
     assert_nil contx['poutine']
   end
+
+  def test_add_global_filter
+    global_filter_proc = ->(output) { "#{output} filtered" }
+
+    context = Context.new
+    context.global_filter = global_filter_proc
+    assert_equal 'hi filtered', context.apply_global_filter('hi')
+
+    context = Context.new
+    assert_equal 'hi', context.apply_global_filter('hi')
+  end
+
+  def test_add_global_filter_when_it_contains_several_actions
```

Comment:
I don't think this test is necessary anymore, since the code above tests the same code path in the code under test.  Doing more in the global filter proc doesn't test anything new about liquid.